source/Calamari.Shared/Deployment/Conventions/StageScriptDependenciesConvention.cs

```diff
@@ -0,0 +1,101 @@
+﻿using System.IO;
+using Calamari.Common.Commands;
+using Calamari.Common.Features.Packages;
+using Calamari.Common.Plumbing.FileSystem;
+using Calamari.Common.Plumbing.Logging;
+using Calamari.Common.Plumbing.Variables;
+using Calamari.Deployment.Conventions.DependencyVariablesStrategies;
+
+namespace Calamari.Deployment.Conventions
+{
+    public class StageScriptDependenciesConvention : IInstallConvention
+    {
+        readonly string? dependencyPathContainingPrimaryFiles;
+        readonly ICalamariFileSystem fileSystem;
+        readonly IPackageExtractor extractor;
+        readonly IDependencyVariablesStrategy dependencyVariablesStrategy;
+        readonly bool forceExtract;
+
+        public StageScriptDependenciesConvention(string? dependencyPathContainingPrimaryFiles, ICalamariFileSystem fileSystem, IPackageExtractor extractor, IDependencyVariablesStrategy dependencyVariablesStrategy, bool forceExtract = false)
+        {
+            this.dependencyPathContainingPrimaryFiles = dependencyPathContainingPrimaryFiles;
+            this.fileSystem = fileSystem;
+            this.extractor = extractor;
+            this.dependencyVariablesStrategy = dependencyVariablesStrategy;
+            this.forceExtract = forceExtract;
+        }
+
+        public void Install(RunningDeployment deployment)
+        {
+            // If the primary files (e.g. a script) is contained in a dependency, then extract the containing dependency in the working directory
+            if (!string.IsNullOrWhiteSpace(dependencyPathContainingPrimaryFiles))
+            {
+                ExtractDependency(dependencyPathContainingPrimaryFiles, deployment.CurrentDirectory);
+                deployment.Variables.Set(KnownVariables.OriginalPackageDirectoryPath, deployment.CurrentDirectory);
```

Comment:
For primary Git dependencies we still use this variable, because they are the same as packages from Calamari's perspective.